feat: add fortune telling app (#62167)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-fortune-teller-app/68e0280810407794171d6558.md

---
id: 68e0280810407794171d6558
title: Step 12
challengeType: 0
dashedName: step-12
---

# --description--

> Now, to make the app more user-friendly, add a div with the class `img-loader`. and after that, create an `img` element and give it a conditional URL for the image using `{img ? ${CDN_URL}/${img} : LOCAL_DEFAULT_IMG}`.

Reviewer: ```suggestion
Now, to make the app more user-friendly, add a div with the class `img-loader`. Next, create an `img` element and give it a conditional URL for the image using `{img ? ${CDN_URL}/${img} : LOCAL_DEFAULT_IMG}`.
```

# --hints--

> Inside the `figure` element you should have `div` element.

Reviewer: ```suggestion
Inside your `figure` element you should have a `div` element.
```

```js
const divInsideFigureRegex = /<figure\b[^>]*>[\s\S]*?<div\b[^>]*>[\s\S]*?<\/div>[\s\S]*?<\/figure>/;
assert.match(code, divInsideFigureRegex)
```

> Your div should have the class of `img-loader`.

Reviewer: ```suggestion
Your div should have a class of `img-loader`.
```

```js
const divClassRegex = /<figure\b[^>]*>[\s\S]*?<div\b[^>]*class\s*=\s*["']img-loader["'][^>]*>[\s\S]*?<\/div>[\s\S]*?<\/figure>/;
assert.match(code, divClassRegex)
```

Inside the figure you should have `img` element.

```js
const imgInsideFigureRegex = /<figure\b[^>]*>[\s\S]*?<img\b[^>]*>[\s\S]*?<\/figure>/;
assert.match(code, imgInsideFigureRegex)
```

> The `img` element should have it's src attribute set to `img ? ${CDN_URL}/${img} : LOCAL_DEFAULT_IMG`.

Reviewer: ```suggestion
The `img` element should have its `src ` attribute set to `img ? ${CDN_URL}/${img} : LOCAL_DEFAULT_IMG`.
```
